osu.Game/Rulesets/Objects/Legacy/LegacyConverterExtensions.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+
+namespace osu.Game.Rulesets.Objects.Legacy
+{
+    public static class LegacyConverterExtensions
+    {
+        public static int GetSplitCount(this ReadOnlySpan<char> text, char separator)
```

Comment:
You can use `MemoryExtensions.Count` here. It's extra fast with full SIMD support for primitive types.